src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -883,7 +884,9 @@ torch::Tensor VideoDecoder::getFramesPlayedInRangeAudio(
         return cursor_ < avFrame->pts + getDuration(avFrame);
       });
       auto frameOutput = convertAVFrameToFrameOutput(avFrameStream);
-      tensors.push_back(frameOutput.data);
+      firstFramePtsSeconds =
+          std::min(firstFramePtsSeconds, frameOutput.ptsSeconds);
+      frames.push_back(frameOutput.data);
```

Comment:
It's now a bit ugly that we are manipulating both a FrameOutput and an `AudioFramesOutput` here in this function.

This is mostly because `convertAVFrameToFrameOutput` returns a `FrameOutput`, but maybe we could bypass it and directly call `convertAudioAVFrameToFrameOutputOnCPU`. I'll write a TODO to investigate this.

Reply:
Oh, yeah, that's cumbersome. If we're in an audio-only call, we shouldn't have to deal with video structures.